src/callbacks/memory_profiler.cpp

```diff
@@ -82,19 +86,56 @@ size_t report_dist_matrix(El::AbstractDistMatrix<T> const& m,
   return allocated;
 }
 
+#ifdef LBANN_HAS_DISTCONV
+template <typename T>
+size_t report_distconv_matrix(::lbann::dc::TensorDev<T> const& m,
+                              std::ostream& stream)
+{
+  size_t const allocated = m.get_local_real_size() * sizeof(T);
+  auto const& shp = m.get_shape();
+  auto const& lshp = m.get_local_real_shape();
+  stream << shp[0];
+  for (int i = 1; i < shp.num_dims(); ++i) {
+    stream << " x " << shp[i];
+  }
+  stream << " (local shape (with halo): " << lshp[0];
+  for (int i = 1; i < lshp.num_dims(); ++i) {
+    stream << " x " << lshp[i];
+  }
+  stream << "). Size: " << allocated / 1048576.0 << " MiB" << std::endl;
+  return allocated;
+}
+#else
+template <typename T>
+size_t report_distconv_matrix(T const& m, std::ostream& stream)
+{
+  stream << "Distconv is disabled" << std::endl;
+  return 0;
+}
+#endif
+
 template <typename T>
 size_t get_activation_and_error_signal_size(data_type_layer<T> const& dtl,
                                             std::ostream& reps)
 {
   size_t allocated = 0;
   for (int i = 0; i < dtl.get_num_children(); ++i) {
-    auto const& act = dtl.get_activations(i);
     if (dtl.get_num_children() == 1)
       reps << "    Activations: ";
     else
       reps << "    Activations (" << i << "): ";
 
-    allocated += report_dist_matrix(act, reps);
+    if (dtl.distconv_enabled()) {
+#ifdef LBANN_HAS_DISTCONV
+      auto const& child = dtl.get_child_layer(i);
+      auto const& dcact = dtl.get_distconv_adapter().get_activations(child);
+      allocated += report_distconv_matrix(dcact, reps);
+#endif
+    }
+    else {
+      auto const& act = dtl.get_activations(i);
+      allocated += report_dist_matrix(act, reps);
+    }
```

Comment:
I think this might undercount memory usage at the distconv boundary. For example, if a layer has to copy out its H2 tensor to a H1 matrix because the next layer is not distconv enabled.